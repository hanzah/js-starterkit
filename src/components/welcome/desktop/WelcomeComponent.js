import React, { PropTypes } from 'react'
import cn from 'classnames'

import styles from './WelcomeComponent.scss'

const { string } = PropTypes

export default class Welcome extends React.Component {
  render() {
    return (
      <div className={cn(styles.wrapper, styles.color)}>Hello {this.props.name}</div>
    );
  }
}

Welcome.propTypes = {
  name: string
}
