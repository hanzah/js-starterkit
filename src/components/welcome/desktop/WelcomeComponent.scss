@import "~components/welcome/WelcomeComponent";

.color {
	color: $base-color;
}
