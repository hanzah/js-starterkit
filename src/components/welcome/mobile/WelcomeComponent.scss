@import "~components/Welcome/WelcomeComponent";

.wrapper {
	width: 100px;
}
