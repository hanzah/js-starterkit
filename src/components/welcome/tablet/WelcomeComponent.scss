@import "~components/welcome/WelcomeComponent";

.green {
	color: green;
}
