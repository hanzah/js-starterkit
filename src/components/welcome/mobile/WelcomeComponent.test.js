import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import WelcomeComponent from './WelcomeComponent'

describe('Welcome (Snapshot)', () => {
  it('Welcome renders hello world', () => {
    const component = renderer.create(<WelcomeComponent name="world"/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

	it('Welcome renders hello world with Enzyme', () => {
		const welcome = shallow(<WelcomeComponent name="world"/>);
    expect(welcome.find('div').text()).toBe('Hello world');
  });
});
