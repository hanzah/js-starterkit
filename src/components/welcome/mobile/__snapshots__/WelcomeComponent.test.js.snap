exports[`Welcome (Snapshot) Welcome renders hello world 1`] = `
<div
  className="">
  Hello 
  world
</div>
`;
